hal/avr32/sonar_i2cxl.c

```diff
@@ -96,13 +98,41 @@ void sonar_i2cxl_get_last_measure(sonar_i2cxl_t* sonar_i2cxl)
 	
 	if ( distance_m > sonar_i2cxl->data.min_distance && distance_m < sonar_i2cxl->data.max_distance )
 	{
+		dt = ((float)time_us - sonar_i2cxl->data.last_update)/1000000.0f;
+
+		if (sonar_i2cxl->data.healthy)
+		{
+			velocity = (distance_m - sonar_i2cxl->data.current_distance) / dt;
+			if (abs(velocity)>20.0f)
```

Comment:
@ndousse 
You should better comment here what you are doing...
I guess you are casting the sonar velocity to remain in a plausible range, isn't it ?
If yes,
Do you really think that the quad can stably reach 20m/s vertical speed ?